### PR TITLE
fix(ci,tests): improve test failure screenshot files naming

### PR DESCRIPTION
**Pull request by odockal**
### What issues does this PR fix or reference?
#4905

### How to test this PR?
Include some invalid character (as defined by filesystem or third party program) into test method name, make the test fail and check failure screenshot file name in `tests/output/screenshots` folder.

### CODE REVIEW

File: tests/src/setupFiles/extended-hooks.ts [hunk 1]
```diff
@@ -23,11 +23,7 @@ import * as path from 'node:path';
 
 afterEach(async (context: RunnerTestContext) => {
   context.onTestFailed(async () => {
-    const normalizedFilePath = context.task.name
-      .replace(/'/g, '')
-      .replace(/\//g, '_')
-      .replace(/:/g, '_')
-      .replace(/ /g, '_');
+    const normalizedFilePath = context.task.name.replace(/([/: ])/g, '_').replace(/(['"><|*?\r\n])/g, '');
```
Comment (benoitf) [inline tests/src/setupFiles/extended-hooks.ts]: I was wondering if we should not like replace all non letters/digits characters ?

something like `replace(/[^a-zA-Z0-9]/g, '')`;

Comment (jeffmaury) [reply tests/src/setupFiles/extended-hooks.ts]: Seems more readable

Comment (odockal) [reply tests/src/setupFiles/extended-hooks.ts]: I have included a bit more sophisticated replacement that will remove most characters that are not letters/numbers or underscore, which I want to use as a delimiter to a words in the filename to keep better readability.

**Review by jeffmaury — APPROVED** [review]
LGTM

**Review by cdrage — APPROVED** [review]
LGTM hopefully this fixes some of the flakes

**Pull request merged**